clientv3/watch.go

```diff
@@ -392,9 +395,11 @@ func (w *watcher) serveStream(ws *watcherStream) {
 				closing = true
 				break
 			}
-			newRev := wrs[0].Events[len(wrs[0].Events)-1].Kv.ModRevision
-			if newRev != ws.lastRev {
-				ws.lastRev = newRev
+			if len(wrs[0].Events) > 0 {
+				newRev := wrs[0].Events[len(wrs[0].Events)-1].Kv.ModRevision
+				if newRev != ws.lastRev {
+					ws.lastRev = newRev
```

Comment:
Progress notification sends no events. Then how do we get the mod revision of the events to update the `ws.lastRev`? Update with `ws.Header.Revision`? Thanks!